src/compiler/checker.ts

```diff
@@ -23713,9 +23713,9 @@ namespace ts {
             }
         }
 
-        function checkArithmeticOperandType(operand: Node, type: Type, diagnostic: DiagnosticMessage): boolean {
+        function checkArithmeticOperandType(operand: Node, type: Type, diagnostic: DiagnosticMessage, tryAwait = false): boolean {

```

Comment:
I'm not sold on the parameter name. Seems more like 'suggestAwait' or something.

Reply:
Whether `await` is actually suggested is left up to some additional checks performed in the function; it could be `trySuggestAwait` or `maybeSuggestAwait` or `isAwaitValid`. I think the last sounds best. Thanks 👍